tiledb/sm/filesystem/s3.cc

```diff
@@ -2029,9 +2037,14 @@ Status S3::get_make_upload_part_req(
   if (!success) {
     UniqueReadLock unique_rl(&multipart_upload_rwlock_);
     auto state = &multipart_upload_states_.at(uri_path);
-    Status st = Status_S3Error(
-        std::string("Failed to upload part of S3 object '") + uri.c_str() +
-        outcome_error_message(upload_part_outcome));
+    auto msg = std::string("Failed to upload part of S3 object '") +
+               uri.c_str() + "' " + outcome_error_message(upload_part_outcome);
+    if (ctx.upload_part_num > max_multipart_part_num) {
+      msg +=
+          " This error might be resolved by increasing the value of the "
```

Comment:
I would phrase it as: `Maximum part limit might have been hit, try increasing the value of "'vfs.s3.multipart_part_size' config option`

Reply:
I think that the fact that a part limit has been reached is effectively conveyed in the `Part number must be an integer between 1 and 10000, inclusive` message returned by S3.